libslide/src/common.rs

```diff
@@ -3,14 +3,28 @@
 /// Describes the character span of a substring in a text.
 ///
 /// For example, in "abcdef", "bcd" has the span (1, 4).
-#[derive(PartialEq, Eq, Copy, Clone, Debug)]
+#[derive(PartialEq, Eq, Copy, Clone, Debug, Hash)]
 pub struct Span {
     /// Inclusive lower bound index of the span
     pub lo: usize,
     /// Exclusive upper bound index of the span
     pub hi: usize,
 }
 
+/// A dummy span for use in places where a span is not (yet) known.
+/// Clearly this span is incorrect since lo = 10001 > 1 = hi, but a well-formed span must observe
+/// the invariant lo <= hi.
+pub(crate) static DUMMY_SP: Span = Span { lo: 10001, hi: 1 };
+
```

Comment:
this seems bad.

Reply:
I don't understand this I think, can you explain. I see where you use it for the tests, but don't you think writing the macro to take an option here could be a better route and then if none is passed, you can just use it. Basically were like emulating optional params in rust.

Reply:
I think it is better to be explicit, otherwise it will look like we are passing "no" span to an expression, even though we always have to have one. Also this way it is easier to find all missing spans since we can grep for DUMMY_SP. I will add a comment that this should not be used in general outside of refactoring. Lmk what you think.

Reply:
that sounds good